Tests/SonarQube.MSBuild.Tasks.IntegrationTests/TargetsTests/RoslynTargetsTests.cs

```diff
@@ -585,10 +606,13 @@ private static void AssertCodeAnalysisIsDisabled(BuildResult result)
 
         private static void AssertExpectedAnalysisProperties(BuildResult result, string expectedErrorLog, string expectedResolvedRuleset, string expectedAdditionalFilesElement)
         {
-            // Check the ruleset and error log are not set
-            BuildAssertions.AssertExpectedPropertyValue(result.ProjectStateAfterBuild, TargetProperties.ErrorLog, expectedErrorLog);
-            BuildAssertions.AssertExpectedPropertyValue(result.ProjectStateAfterBuild, TargetProperties.ResolvedCodeAnalysisRuleset, expectedResolvedRuleset);
```

Comment:
I factored all of the checks in this method out into separate methods so this method is now just a convenient wrapper for the case when all three values are known.